src/index.ts

```diff
@@ -1,8 +1,13 @@
 import { readdirSync } from 'fs';
 import { join, parse } from 'path';
+import { TSESLint } from '@typescript-eslint/experimental-utils';
 import globals from './globals.json';
 import * as snapshotProcessor from './processors/snapshot-processor';
 
+type RuleModule = TSESLint.RuleModule<string, unknown[]> & {
```

Comment:
should we use this type for all rules where we declare them as well?

Reply:
How do you mean?

Ideally we should try and use the inferred type that's returned from `createRule` as much as possible for each rule, as that has the proper types for their message ids & options.

We could switch to using a barrel export like `@typescript-eslint` does, where we have `index.ts` that explicitly exports each rule:

```
import noIfs from './no-ifs';

export default  {
  'no-ifs': noIfs,
}
```

That would let us import all the rules here as a single default import.

<details>
<summary>
we can even script it
</summary>

```
#!/usr/bin/env ts-node-transpile-only

import * as fs from 'fs';
import * as path from 'path';
import prettier from 'prettier';
import { prettier as prettierConfig } from '../package.json';

const toCamelCase = (str: string) => {
  const words = str.split('-');

  return [
    words.shift(),
    ...words.map(w => w.substring(0, 1).toUpperCase() + w.substring(1)),
  ].join('');
};

const excludedFiles = ['__tests__', 'utils', 'index'];

const ruleFiles = fs
  .readdirSync(path.resolve(__dirname, '../src/rules'))
  .map(file => path.parse(file).name)
  .filter(file => !excludedFiles.includes(file));

const contents = [
  ...ruleFiles.map(rule => `import ${toCamelCase(rule)} from './${rule}';`),
  '',
  'export default {',
  ...ruleFiles.map(rule => `  '${rule}': ${toCamelCase(rule)},`),
  '};', //
].join('\n');

fs.writeFileSync(
  path.resolve(__dirname, '../src/rules/index.ts'),
  prettier.format(contents, { parser: 'typescript', ...prettierConfig }),
);
```
</details>

Reply:
(I'm going to merge this, but happy to continue the discussion 🙂 )

Reply:
I meant doing something here so we use the same type: https://github.com/jest-community/eslint-plugin-jest/blob/75f8897944059baffd80d665e3d9cbd5641d46b1/src/rules/utils.ts#L12-L16
Not sure how feasible it is, tho?